lodash.js

```diff
@@ -2568,17 +2568,21 @@
      * @param {boolean[]} orders The sort orders of `props`.
      * @returns {Array} Returns the new sorted array.
      */
-    function baseSortByOrder(collection, props, orders) {
+    function baseSortByOrder(collection, comparitors, orders) {
       var index = -1,
           length = collection.length,
           result = isLength(length) ? Array(length) : [];
 
+      comparitors = arrayMap(comparitors, function(comparitor) {
+        return getCallback(comparitor, null, 1);
+      });
+
       baseEach(collection, function(value) {
         var length = props.length,
             criteria = Array(length);
 
         while (length--) {
-          criteria[length] = value == null ? undefined : value[props[length]];
+          criteria[length] = comparitors[length](value);
         }
```

Comment:
Works for me
On Apr 12, 2015 12:46 PM, "John-David Dalton" <notifications@github.com>
wrote:

> In lodash.js
> <https://github.com/lodash/lodash/pull/1126#discussion_r28205118>:
>
> >        baseEach(collection, function(value) {
> >          var length = props.length,
> >              criteria = Array(length);
> >
> >          while (length--) {
> > -          criteria[length] = value == null ? undefined : value[props[length]];
> > +          criteria[length] = comparitors[length](value);
> >          }
>
>  Yep, that was the other thing - should context param be added
>
> The thing is both methods can accept multiple prop names (potential
> iteratees) so the binding would apply to all of them? That gets messy. I
> think that using _.bind in this case would be better.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lodash/lodash/pull/1126/files#r28205118>.
>